src/libostree/ostree-sysroot-cleanup.c

```diff
@@ -555,6 +555,48 @@ ostree_sysroot_cleanup (OstreeSysroot       *self,
   return _ostree_sysroot_cleanup_internal (self, TRUE, cancellable, error);
 }
 
+/**
+ * ostree_sysroot_auto_cleanup:
+ * @self: Sysroot
+ * @out_cleaned: (out): Whether a cleanup was performed
+ * @cancellable: Cancellable
+ * @error: Error
+ *
+ * Cleanup @self when needed. Whether to cleanup or not is determined by
+ * the presence of the .cleanup file in the sysroot. When the .cleanup
+ * file exists, ostree_sysroot_cleanup() will be called and the file
+ * will be removed if it is successful.
+ *
+ * Locking: exclusive
+ * Since: 2022.2
+ */
+gboolean
+ostree_sysroot_auto_cleanup (OstreeSysroot       *self,
+                             gboolean            *out_cleaned,
+                             GCancellable        *cancellable,
+                             GError             **error)
+{
+  struct stat stbuf;
+  if (!glnx_fstatat_allow_noent (self->sysroot_fd, _OSTREE_SYSROOT_AUTOCLEANUP, &stbuf, AT_SYMLINK_NOFOLLOW, error))
```

Comment:
Since the systemd unit is already doing this, do we need a new API at all versus simply `ExecStart=ostree admin cleanup`?

Reply:
Maybe not. I think you could do this all in the CLI or via systemd unit settings. It felt cleaner to encapsulate it in libostree, but it's not necessary. I could go either way.